Handler/GelfHandlerWrapper.php

```diff
@@ -18,15 +20,60 @@
  */
 class GelfHandlerWrapper extends GelfHandler
 {
+    use VerifyConfigurationTrait;
+
+    /**
+     * @var string
+     */
+    private $isEnabled;
+
+    /**
+     * @var string
+     */
+    private $levelPath;
+
+    /**
+     * @var ScopeConfigInterface
+     */
+    private $scopeConfig;
+
+    /**
+     * @var GelfHandler
+     */
+    private $gelfHandler;
+
     /**
      * GelfHandlerWrapper constructor.
      *
-     * @param PublisherInterface $publisher
-     * @param int                $level
-     * @param bool               $bubble
+     * @param PublisherInterface   $publisher
+     * @param ScopeConfigInterface $scopeConfig
+     * @param string               $isEnabled
+     * @param string               $levelPath
      */
-    public function __construct(PublisherInterface $publisher, $level = Logger::DEBUG, $bubble = true)
+    public function __construct(
+        PublisherInterface $publisher,
+        ScopeConfigInterface $scopeConfig,
+        string $isEnabled,
+        string $levelPath
+    ) {
+        $this->publisher = $publisher;
+        $this->isEnabled = $isEnabled;
+        $this->levelPath = $levelPath;
+        $this->scopeConfig = $scopeConfig;
+    }
+
+    /**
+     * @return GelfHandler
+     */
+    final public function getInstance(): GelfHandler
     {
-        parent::__construct($publisher, $level, $bubble);
+        if (!$this->gelfHandler) {
+            $this->gelfHandler = new GelfHandler(
```

Comment:
Même remarque.

J'ai l'impression que tu veux faire un singleton. Mais qui ne marche que si ta classe a déjà été instanciée puisque le getInstance a besoin de propriétés de `$this`.

Ou alors c'est le extends qui est plus pertinent et qui doit sauter ? Puisque d'ailleurs on appelle plus le `__construct`du parent.